src/common/buffer.cc

```diff
@@ -1402,6 +1402,28 @@ using namespace ceph;
     }
   }
 
+  uint64_t buffer::list::get_wasted_space() const
+  {
+    if (_buffers.size() == 1)
+      return _buffers.back().wasted();
+
+    std::vector<const raw *> raw_vec;
+    raw_vec.reserve(_buffers.size());
+    for (const auto& p : _buffers)
+      raw_vec.push_back(p.get_raw());
+    std::sort(raw_vec.begin(), raw_vec.end());
+
+    uint64_t total = 0;
+    const raw *last = nullptr;
+    for (const auto r : raw_vec) {
```

Comment:
please note, if multiple `ptr`s are sharing the same `raw` buffer and they overlap with each other, we will underestimate the wasted space. as the `length()` takes the overlapped spaces into consideration. probably this does not happen in the usecase of `ObjectCacher`, it might be worthy to mention as a comment in this method though.